pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -399,12 +437,40 @@ func (c *Controller) Stop() {
 
 // Services implements a service catalog operation
 func (c *Controller) Services() ([]*model.Service, error) {
-	c.RLock()
+	var k8sNodes []*kubernetesNode
+	var extAddresses []string
+	rawNodes := c.nodes.GetStore().List()
+	for _, i := range rawNodes {
```

Comment:
This is a hot point, every time we get service, it will iterate all the nodes.

Could introduce a specific label indicate the node is accessible from remote cluster and cache it.

Reply:
Also you need to watch node events and trigger push accordingly.